### PR TITLE
Feat: Make vim fully run on gpu

### PR DESCRIPTION
**Pull request by y-prudent**
## Features:

- VIM now fully runs on GPU (no more heavy operation in numpy)
- Delete `KneeLocator` that used to find an elbow point on the cumulated explained variance curve to get a default number of principal dims. It was a nice feaure but unfortunately sometimes returned a None value, raising an error. Plus it has not been sufficiently tested. Now, default value for `princ_dims` is 0.99 (meaning that the number of principal dims is chosen so that they cover 99% of explained variance).
- Clearer `plot_spectrum` function by plotting the cumulated explained variance for residual space instead of principal space. 

Ready to be reviewed!

### CODE REVIEW

**Review by cofri — APPROVED** [review]
:rocket:

File: oodeel/methods/vim.py [hunk 1]
```diff
@@ -199,31 +157,11 @@ def _compute_residual_score_tensor(self, features: TensorType) -> TensorType:
         Returns:
             scores
```
Comment (paulnovello) [inline oodeel/methods/vim.py:158]: Add scores type

**Review by paulnovello — CHANGES_REQUESTED** [review]
Some type annotation remarks 
(sorry should have been addressed by me in the last PR but *errare humanum est*)

File: oodeel/methods/base.py [hunk 2]
```diff
@@ -54,7 +55,7 @@ def __init__(
         output_layers_id: List[Union[int, str]] = [-1],
         input_layers_id: Optional[Union[int, str]] = None,
     ):
-        self.feature_extractor = None
+        self.feature_extractor: FeatureExtractor = None
```
[the remark this answers is not in the document]
Comment (y-prudent) [reply oodeel/methods/base.py:58]: By expliciting the type of `self.feature_extractor`, vscode (or any other IDE) is now able to display the documentation of the feature extractor methods when hovering mouse on the `feature_extractor` attribute of an OODBaseDetector object. I personnally find this very useful while coding a new baseline.

**Review by paulnovello — CHANGES_REQUESTED** [review]

**Comment by paulnovello** [comment]
LGTM!

**Pull request merged**